sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -21,13 +21,20 @@ public class ServiceBusConnectionStringBuilder
         const string SharedAccessKeyNameConfigName = "SharedAccessKeyName";
         const string SharedAccessKeyConfigName = "SharedAccessKey";
         const string SharedAccessSignatureConfigName = "SharedAccessSignature";
+        const string AuthenticationConfigName = "Authentication";
 
         const string EntityPathConfigName = "EntityPath";
         const string TransportTypeConfigName = "TransportType";
 
         const string OperationTimeoutConfigName = "OperationTimeout";
 
         string entityPath, sasKeyName, sasKey, sasToken, endpoint;
+        string authType;
+
+        public static class AuthenticationType
```

Comment:
I'm not sure that this is really discoverable tucked in here as a nested class.   How would we expect callers to know to look inside the connection string builder to find the authentication type constants?

Reply:
Feels a bit redundant to make it its separate class, especially with just one value right now. What do you think about just using a public const like "AuthenticationManagedIdentity"?

---
In reply to: [292684143](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r292684143) [](ancestors = 292684143)

Reply:
From the perspective of a consumer of the API, that does not know about this particular magic string nor the implementation...  can you describe the scenario for how you would expect them to discover this value?   I'm not clear on how a consumer would know to explore the nested class or the `ConnectionStringBuilder` members to discover it.

The reason that I suggest a dedicated type is that if I look in intellisense and see a class called `AuthenticationType` or similar and explore the members, I'm more likely to be able to discover it.  Frankly, that is the reason that I'm pushing for a true enum here.  The type name of the enum in the message signature clearly tells callers where to look for the valid set of values with no intuitive leap.

Reply:
To be honest, I'm not too concerned about the discovery of this value to external users of this API. It's very unlikely that the user would need to manually do anything with this value, since the connection string is just a generated value that they need to copy+paste from elsewhere and feed as an input to this API. I think having this value being discoverable/convenient to other developers is probably more important.

---
In reply to: [293031840](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r293031840) [](ancestors = 293031840)

Reply:
Anything that is public is users concern. Otherwise, it should be internal.
Also, "users" are developers as well 🙂